### PR TITLE
48090 gui categories backend

### PR DESCRIPTION
**Pull request by shawon-sharma**
Here i have just fetch the categories name wit id

### CODE REVIEW

File: app/src/main/res/layout/categories_list.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        android:textSize="35sp"
+        android:textColor="#000000"
+        android:textStyle="bold"
+        android:gravity="center"/>
+    <TextView
+        android:id="@+id/textBody"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text=""
+        android:textSize="25sp"
```
Comment (nvmanh) [inline app/src/main/res/layout/categories_list.xml]: hardcode

File: app/src/main/res/layout/categories_list.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    android:orientation="vertical"
+    android:layout_height="match_parent"
+    android:background="@drawable/category_row_border">
+    <TextView
+        android:id="@+id/textHeader"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text=""
+        android:textSize="35sp"
+        android:textColor="#000000"
```
Comment (nvmanh) [inline app/src/main/res/layout/categories_list.xml]: hardcode

File: app/src/main/res/drawable/category_row_border.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <stroke android:width="5dp" android:color="#3b5998" />
```
Comment (nvmanh) [inline app/src/main/res/drawable/category_row_border.xml]: hardcode

File: app/src/main/java/com/framgia/elsytem/QuestionActivity.java [hunk 4]
```diff
@@ -215,73 +185,13 @@ public void onInit(int status) {
             Log.e("TTS", "Initilization Failed!");
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/QuestionActivity.java]: please return, do not use else case with nothing

File: app/src/main/java/com/framgia/elsytem/QuestionActivity.java [hunk 5]
```diff
@@ -133,34 +118,19 @@ public void createalert() {
                     @Override
                     public void onClick(DialogInterface dialog, int which) {
                         dialog.cancel();
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/QuestionActivity.java]: break this to new method, it so long

File: app/src/main/java/com/framgia/elsytem/LoginActivity.java [hunk 6]
```diff
@@ -200,7 +205,12 @@ protected void onPostExecute(String result) {
                         session.createLoginSession(id, name, email, avatar, authToken,
                                 mRememberMe);
                     //now finish this activity and go to ProfileActivity
-                    Intent i = new Intent(getApplicationContext(), ProfileActivity.class);
+                    /*Intent i = new Intent(getApplicationContext(), ProfileActivity.class);
+                    startActivity(i);
+                    finish();*/
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/LoginActivity.java]: remove all dummy text

File: app/src/main/java/com/framgia/elsytem/LoginActivity.java [hunk 7]
```diff
@@ -54,7 +54,12 @@ protected void onCreate(Bundle savedInstanceState) {
         // check in session if user is logged in. If so, go to profile activity
         if (session.isLoggedInAndRemember()) {
             //go to ProfileActivity
-            Intent i = new Intent(getApplicationContext(), ProfileActivity.class);
+            /*Intent i = new Intent(getApplicationContext(), ProfileActivity.class);
+            startActivity(i);
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/LoginActivity.java]: remove all these

File: app/src/main/java/com/framgia/elsytem/CategoriesActivity.java [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        @Override
+        protected ArrayList<CategoryResponse.CategoriesEntity> doInBackground(String... urls) {
+            String json = null;
+            try {
+                Log.e("token ", token);
+                json = createjson(token);
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+            RequestBody requestBody = RequestBody.create(JSON, json);
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/CategoriesActivity.java]: if json null, we still continue???. Why you use 2 try-catch block in this method

File: app/src/main/java/com/framgia/elsytem/CategoriesActivity.java [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+        token = user.get(mConstant.KEY_AUTH_TOKEN);
+        category = (TextView) findViewById(R.id.textHeader);
+        list = (ListView) findViewById(R.id.listCategory);
+        okHttpClient = new OkHttpClient();
+        new HttpAsyncCategory().execute("https://manh-nt.herokuapp.com/categories.json?auth_token=" + user.get(mConstant.KEY_AUTH_TOKEN));
+    }
+
+    public String createjson(String token) throws JSONException {
+        JSONObject object = new JSONObject();
+        object.put("page", 1);
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/CategoriesActivity.java]: create constant values please

File: app/src/main/java/com/framgia/elsytem/CategoriesActivity.java [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        setContentView(R.layout.activity_categories);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        sessionManager = new SessionManager(this);
+        user = sessionManager.getUserDetails();
+        token = user.get(mConstant.KEY_AUTH_TOKEN);
+        category = (TextView) findViewById(R.id.textHeader);
+        list = (ListView) findViewById(R.id.listCategory);
+        okHttpClient = new OkHttpClient();
+        new HttpAsyncCategory().execute("https://manh-nt.herokuapp.com/categories.json?auth_token=" + user.get(mConstant.KEY_AUTH_TOKEN));
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/elsytem/CategoriesActivity.java]: constant please

File: app/src/main/java/com/framgia/elsytem/QuestionActivity.java (hunk 5 above)
Comment (shawon-sharma) [inline app/src/main/java/com/framgia/elsytem/QuestionActivity.java]: this activity class was written by Ahsan -san..i will tell him to change it in his next pull ..

File: app/src/main/java/com/framgia/elsytem/QuestionActivity.java (hunk 4 above)
Comment (shawon-sharma) [inline app/src/main/java/com/framgia/elsytem/QuestionActivity.java]: this activity class was written by Ahsan -san..i will tell him to change it in his next pull ..

File: app/src/main/res/layout/result.xml [hunk 11]
```diff
@@ -18,14 +18,12 @@
         android:layout_height="wrap_content"
         android:text="@string/result_word_name_text"
         android:layout_marginLeft="15dp"
```
Comment (nvmanh) [inline app/src/main/res/layout/result.xml]: hardcode

**Pull request merged**